datalad/interface/tests/test_run.py

```diff
@@ -143,6 +143,12 @@ def test_rerun(path, nodspath):
     # ran twice now
     eq_('xx\n', open(probe_path).read())
 
+    # Rerunning from a subdataset skips the command.
+    assert_result_count(
+        sub.rerun(return_type="list", on_failure="ignore"),
+        1, status="impossible", action="run", rerun_action="skip")
+    eq_('xx\n', open(probe_path).read())
```

Comment:
> to have correct said recorded within subdataset

Sorry, I can't parse that.

> a dedicated unit test

I piggybacked on the rerun test that already sets up and tests things about subdatasets to avoid duplicating similar setup and adding unnecessary time to these tests.